### PR TITLE
[aws][fix] Collect and connect Inspector resources properly

### PR DESCRIPTION
**Pull request by 1101-1**
# Description
fix connection issues

# To-Dos

- [x] I have created tests for any new or updated functionality.
- [x] I ran `tox` successfully.

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://inventory.fix.security/code-of-conduct).

### CODE REVIEW

File: plugins/aws/fix_plugin_aws/resource/inspector.py [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+                            self,
+                            clazz=AwsLambdaFunction,
+                            arn=lambda_arn,
+                        )
+                    else:
+                        builder.add_edge(
+                            self,
+                            clazz=(AwsEc2Instance, AwsEcrRepository),
+                            id=rid,
+                        )
```
Comment (aquamatthias) [inline plugins/aws/fix_plugin_aws/resource/inspector.py:443]: based on the arn, you also know if it is ec2 instance or ecr repo, or?

Comment (1101-1) [reply plugins/aws/fix_plugin_aws/resource/inspector.py:443]: Same logic was in 
```python
class AwsInspectorV2Coverage(AwsResource):
    def connect_in_graph(self, builder: GraphBuilder, source: Json) -> None:
        if self.id:
            builder.add_edge(
                self,
                clazz=AwsEc2Instance,
                id=self.id,
            )
        if resource_metadata := self.resource_metadata:
            if lambda_metadata := resource_metadata.lambda_function:
                builder.add_edge(
                    self,
                    clazz=AwsLambdaFunction,
                    id=lambda_metadata.function_name,
                )
            if ecr_metadata := resource_metadata.ecr_repository:
                builder.add_edge(
                    self,
                    clazz=AwsEcrRepository,
                    id=ecr_metadata.name,
                )
 ```
 For `AwsInspectorFinding` we have this logic:
 If `finding_resource.type` is Ec2 instance, then finding_resource.id is an ID of `Ec2Instance`. If `finding_resource.type` is EcrRepository, then `finding_resource.id` is the name of `EcrRepository`. For lambda `finding_resource.id` will be `arn + lambda_version` of `AwsLambdaFunction`

Comment (aquamatthias) [reply plugins/aws/fix_plugin_aws/resource/inspector.py:443]: So then: let's define the class to filter more explicitly.

**Comment by aquamatthias** [comment]
Let's use the new assessment structure to report inspector findings instead of separate resources.

File: plugins/aws/fix_plugin_aws/resource/base.py [hunk 2]
```diff
@@ -469,7 +503,19 @@ def __init__(
         self.last_run_started_at = last_run_started_at
         self.created_at = utc()
         self.__builder_cache = {region.safe_name: self}
-
+        self._assessment_findings: Dict[Tuple[str, str, str], Dict[str, List[Finding]]] = defaultdict(
```
Comment (aquamatthias) [inline plugins/aws/fix_plugin_aws/resource/base.py:506]: Please see GraphBuilder.for_region - you need to pass the findings.

**Review by aquamatthias — CHANGES_REQUESTED** [review]

**Review by aquamatthias — APPROVED** [review]

**Pull request merged**